src/test/storage/table_test.cpp

```diff
@@ -102,9 +101,47 @@ TEST_F(StorageTableTest, AppendNullValues) {
   EXPECT_THROW(table.append({NULL_VALUE, "foo"}), std::logic_error);
 }
 
+TEST_F(StorageTableTest, CompressChunkMultithreading) {
+  const auto number_columns = ColumnID{100};
+  const auto chunk_size = ChunkOffset{1000};
+
+  const auto col_id_to_name = [](ColumnID col_id) { return "col_" + std::to_string(col_id); };
+
+  auto large_table = Table{chunk_size};
+  auto data_copy = std::vector<std::vector<AllTypeVariant>>();
+  // Add columns.
+  for (auto col_id = ColumnID{0}; col_id < number_columns; ++col_id) {
+    large_table.add_column(col_id_to_name(col_id), "int", false);
+  }
+
+  // Add rows.
+  for (auto row_id = ChunkOffset{0}; row_id < chunk_size; ++row_id) {
+    auto row = std::vector<AllTypeVariant>();
+    row.reserve(number_columns);
+    for (auto col_id = ColumnID{0}; col_id < number_columns; ++col_id) {
+      auto val = rand() % 100;  // NOLINT
+      row.push_back(val);
+    }
+    large_table.append(row);
+    data_copy.push_back(row);
+  }
+
+  large_table.compress_chunk(ChunkID{0});
+
+  const auto column_names = large_table.column_names();
+  const auto chunk = large_table.get_chunk(ChunkID{0});
+
+  // Checking if compressed chunk and data copy are still equal.
+  for (auto col_id = ColumnID{0}; col_id < number_columns; ++col_id) {
+    auto compressed_column = chunk->get_segment(col_id);
+    for (auto row_id = ChunkOffset{0}; row_id < chunk_size; ++row_id) {
+      EXPECT_EQ(data_copy[row_id][col_id], (*compressed_column)[row_id]);
+    }
+  }
+}
+
 TEST_F(StorageTableTest, CompressChunk) {
-  // Not implemented yet
-  EXPECT_ANY_THROW(table.compress_chunk(ChunkID{0}));
+  table.compress_chunk(ChunkID{0});
```

Comment:
What is the purpose of this test? Just checking that compress_chunk doesn't fail or throw exceptions?